File: geo-service/src/main/java/ma/emsi/geoservice/service/GeoService.java
package ma.emsi.geoservice.service;

import ma.emsi.geoservice.dto.PlaceDto;
import ma.emsi.geoservice.dto.PlaceDetailsDto;

import java.util.List;

public interface GeoService {

    PlaceDetailsDto getPlaceDetails(String placeId);

    List<PlaceDto> searchNearby(double lat, double lng, int radius, String keyword);

    double calculateDistanceKm(double lat1, double lon1, double lat2, double lon2);
}

File: restaurant-frontend/src/services/geoService.js
import api from './api';

const BASE_URL = '/geo-service/api/geo';

export const geoService = {
    // Get place details by Google Place ID
    getPlaceDetails: async (placeId) => {
        const response = await api.get(`${BASE_URL}/place/${placeId}`);
        return response.data;
    },

    // Search for nearby places
    searchNearby: async (lat, lng, radius = 1000, keyword = null) => {
        const params = { lat, lng, radius };
        if (keyword) params.keyword = keyword;
        const response = await api.get(`${BASE_URL}/nearby`, { params });
        return response.data;
    },

    // Calculate distance between two points in kilometers
    calculateDistance: async (lat1, lon1, lat2, lon2) => {
        const response = await api.get(`${BASE_URL}/distance`, {
            params: { lat1, lon1, lat2, lon2 }
        });
        return response.data;
    }
};
